src/transport/raw/TCP.cpp

```diff
@@ -39,11 +39,11 @@ namespace {
 
 using namespace chip::Encoding;
 
-// Packets start with a 16-bit size
-constexpr size_t kPacketSizeBytes = 2;
+// Packets start with a 32-bit size field.
+constexpr size_t kPacketSizeBytes = sizeof(uint32_t);
 
-// TODO: Actual limit may be lower (spec issue #2119)
-constexpr uint16_t kMaxMessageSize = static_cast<uint16_t>(System::PacketBuffer::kMaxSizeWithoutReserve - kPacketSizeBytes);
+constexpr uint32_t kMaxTCPMessageSize =
+    static_cast<uint32_t>(System::PacketBuffer::kLargeBufMaxSizeWithoutReserve - kPacketSizeBytes);
```

Comment:
This is making two assumptions that need to be static_asserted:

1. kLargeBufMaxSizeWithoutReserve fits in uint32_t.
2. kLargeBufMaxSizeWithoutReserve is not smaller than kPacketSizeBytes.

Reply:
> The static_assert for (1) is there in PacketBufferHandle::New().

How does that help _this_ code?  This code is making an assumption.  Someone could easily change the value of kLargeBufMaxSizeWithoutReserve, change the assert in PacketBufferHandle::New and not realize that _this_ code is then broken.

Which is why _this_ code should have asserts about the assumptions it's making, even if other places in the system happen to be making the same assumptions and asserting them already.

Reply:
Ok. Added a static_assert here for 2.